.gitlab-ci.yml

```diff
@@ -647,17 +668,39 @@ release:mender-docs-changelog:
     - git config --global user.name "${GITHUB_USER_NAME}"
     # GITHUB_TOKEN for Github cli authentication
     - export GITHUB_TOKEN=${GITHUB_CLI_TOKEN}
+    - THIS_REPO=$(echo ${GITHUB_REPO_URL} | cut -d '/' -f2)
+    - if [[ "${THIS_REPO}" == "mender-server-enterprise" ]]; then
+        export CHANGELOG_SUFFIX="-enterprise";
+      else
+        export CHANGELOG_SUFFIX="";
+      fi;
   script:
     - git clone https://${GITHUB_USER_NAME}:${GITHUB_BOT_TOKEN_REPO_FULL}@github.com/${GITHUB_CHANGELOG_REPO_URL}
     - cd ${GITHUB_CHANGELOG_REPO_URL#*/}
     - git checkout -b changelog-${CI_JOB_ID}
     - cat ../.docs_header.md > ${CHANGELOG_REMOTE_FILE}
-    - cat ../CHANGELOG.md | grep -v -E '^---' >> ${CHANGELOG_REMOTE_FILE}
+    - cat ../CHANGELOG${CHANGELOG_SUFFIX}.md | grep -v -E '^---' >> ${CHANGELOG_REMOTE_FILE}
     - git add ${CHANGELOG_REMOTE_FILE}
     - |
-      git commit -s -m "chore: add mender-server changelog"
+      git commit -s -m "chore: add $CI_PROJECT_NAME changelog"
     - git push origin changelog-${CI_JOB_ID}
-    - gh pr create --title "Update CHANGELOG.md" --body "Automated change to the CHANGELOG.md file" --base master --head changelog-${CI_JOB_ID}
+    - gh pr create --title "Update CHANGELOG${CHANGELOG_SUFFIX}.md for $CI_PROJECT_NAME" --body "Automated change to the CHANGELOG${CHANGELOG_SUFFIX}.md file" --base master --head changelog-${CI_JOB_ID}
+
+release:mender-docs-changelog:saas:
+  extends: release:mender-docs-changelog
+  variables:
+    CHANGELOG_REMOTE_FILE: "12.Hosted-Mender/docs.md"
+  rules:
+    - if: '$CI_PROJECT_NAME == "mender-server"'
+      when: never
+    - if: $CI_COMMIT_TAG
```

Comment:
```suggestion
    - if: $CI_COMMIT_TAG =~ "^v\d+\.\d+\.\d+-saas[\.0-9a-z+]+$"
```

Reply:
On this point: what we want to do when we publish the Stable on-prem tags? For example: v4.1.0, which contains the same code as v4.1.0-saas.rc1 for example. We want to publish it to hosted Mender also or not?

Reply:
You're right! Maybe we should add a restriction that has a version tag prefix?
Just in case we would create other tags for whatever reason.

Reply:
Yes, I'm going to reuse your suggested change and include `saas` as optional.